### PR TITLE
Undo deletion

### PR DESCRIPTION
**Pull request by jacobrs**
#91 Still need to add tests and weird adapter bug to fix.

### CODE REVIEW

File: test/androidTest/java/org/thoughtcrime/securesms/espressoTests/ConversationDeleteTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    public void deleteConversationAndUndoTest() throws InterruptedException {
+
+        String convoRecipient = "+1119991112";
+
+        ConversationActions.createNewConversationWithImmediateExit(convoRecipient, "hello");
+        Expectations.checkIsDisplayed(R.id.from, convoRecipient);
+
+        ConversationActions.deleteConversation(convoRecipient);
+        Thread.sleep(1000);
+        ConversationActions.undoDeletion();
```
Comment (benrs) [inline test/androidTest/java/org/thoughtcrime/securesms/espressoTests/ConversationDeleteTest.java]: Maybe add some sleep after this, cause even if this action didn't do anything I think the test would still pass.

File: test/androidTest/java/org/thoughtcrime/securesms/espressoTests/PersistentMessageStateTest.java [hunk 2]
```diff
@@ -31,6 +30,7 @@
 
     // These tests should only run manually as you need to actually be verified
     //@Test
+    @Ignore
```
Comment (benrs) [inline test/androidTest/java/org/thoughtcrime/securesms/espressoTests/PersistentMessageStateTest.java:33]: Good stuff, wasn't aware of these annotations

File: test/androidTest/java/org/thoughtcrime/securesms/util/Expectations.java [hunk 3]
```diff
@@ -20,21 +20,25 @@ public static void checkIsDisplayed(int id) {
     }
 
     public static void checkIsDisplayed(int id, String message) {
-        onView(allOf(withId(id), withText(message))).check(matches(isDisplayed()));
+        onView(EspressoUtil.withIndex(allOf(withId(id), withText(message)), 0)).check(matches(isDisplayed()));
```
Comment (benrs) [inline test/androidTest/java/org/thoughtcrime/securesms/util/Expectations.java:23]: 👍

File: test/androidTest/java/org/thoughtcrime/securesms/espressoTests/ConversationDeleteTest.java (hunk 1 above)
Comment (jacobrs) [reply test/androidTest/java/org/thoughtcrime/securesms/espressoTests/ConversationDeleteTest.java]: actually I might add a check after the delete to make sure it doesn't exist. then the check will be 100% guaranteed.

**Review by benrs — APPROVED** [review]
Looks good

**Review by samantha-kerr — APPROVED** [review]
Works well! The only thing that didn't work all the time (at least for me) was the UI test; the swipeLeft action sometimes would not swipe enough to delete the conversation. But the test is written properly.

**Comment by jacobrs** [comment]
When I ran the UI test it worked everytime, how often did it fail for you @samantha-kerr ? If its two often I'll refactor the test in another task. Merging for now.

**Pull request merged**